Extension/src/LanguageServer/extension.ts

```diff
@@ -343,6 +344,47 @@ function onToggleSquiggles(): void {
     settings.toggleSetting("errorSquiggles", "Enabled", "Disabled");
 }
 
+function onToggleSnippets(): void {
+    onActivationEvent();
+
+    // This will apply to all clients as it's a global toggle. It will require a reload.
+    const snippetsCatName: string  = "Snippets";
+    const snippetsNodeName: string = "snippets";
+    let newPackageJson: any = util.getRawPackageJson();
+
+    if (newPackageJson.categories.findIndex(cat => cat === snippetsCatName) === -1) {
+        // Add the Snippet category and snippets node. 
+
+        newPackageJson.categories.push(snippetsCatName);
+        newPackageJson.contributes[snippetsNodeName] = [{"language": "cpp", "path": "./cpp_snippets.json"}, {"language": "c", "path": "./cpp_snippets.json"}];
```

Comment:
nit: ```newPackageJson.contributes.snippets```

Same with the associated delete below.

Makes it easier to read the code and know what part of the package.json its modifying.